### DISCUSSION TITLE
Change input payload when using refreshInterval

### DISCUSSION THREAD

**Discussion by bobbylemm** [vercel/swr#1845] [Q&A]
Hello,
I currently use `graphql-request` as my fetcher library, I am trying to make multiple requests to an API using the `refreshInterval` config. This works well right now, is it possible to change the input payload for each unique request to the API because currently it uses the input payload for all requests.

```javascript
import { request } from 'graphql-request'

const fetcher = query => request('/api/graphql', query)

const data = useSWR(`query getLink(
    $email: String!
    $recaptchaToken: String!
  ) {
    signInLinkForEmailAndOTP(
      email: $email
      recaptchaToken: $recaptchaToken
    )
  }`, fetcher, {refreshInterval: 2000})
```
I need to send a new ReCaptcha token in the payload for each request to the API. I do not know if this is possible with this library.

**Comment by icyJoseph** [comment]
Hi,

Kinda curious on how is this working without variables? In any case, you could change fetcher to: `(queryKey) => fetcher(queryKey, getCaptchaValue())` - where the `getCaptchaValue` function makes sure to get the most fresh value for your operation.
